actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -381,7 +381,7 @@ def valid_per_form_csrf_token?(token, session) # :doc:
         if per_form_csrf_tokens
           correct_token = per_form_csrf_token(
             session,
-            request.fullpath.chomp("/"),
+            normalize_action_path(request.fullpath),
```

Comment:
Doesn't `request.path.chomp("/")` work?

Reply:
You are right, it works. I've changed it to `request.path`.